nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/blob/DocumentBlobManagerComponent.java

```diff
@@ -434,13 +450,20 @@ public BinaryManagerStatus garbageCollectBinaries(boolean delete) {
 
     @Override
     public void markReferencedBinary(String key, String repositoryName) {
-        BlobProvider blobProvider = getBlobProvider(key, repositoryName);
-        BinaryGarbageCollector gc = blobProvider.getBinaryGarbageCollector();
-        if (gc != null) {
-            key = stripBlobKeyPrefix(key);
-            gc.mark(key);
+        List<BinaryGarbageCollector> gcs = getGarbageCollectors();
+        Set<String> ids = gcs.stream().map(BinaryGarbageCollector::getId).collect(toSet());
```

Comment:
Could it be interesting to just compute the count of unique ids?

It could be achieved like this:

```suggestion
        long idCount = gcs.stream().map(BinaryGarbageCollector::getId).distinct().count();
```